.cbmc-batch/jobs/aws_cryptosdk_cmm_generate_enc_materials/aws_cryptosdk_cmm_generate_enc_materials_harness.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ * use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on
+ * an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <aws/cryptosdk/materials.h>
+#include <aws/cryptosdk/private/keyring_trace.h>
+#include <cbmc_invariants.h>
+#include <cipher_openssl.h>
+#include <make_common_data_structures.h>
+#include <proof_helpers/cryptosdk/make_common_data_structures.h>
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+#include <proof_helpers/utils.h>
+
+// Stub this until https://github.com/diffblue/cbmc/issues/5344 is fixed
+// Original function is here:
+// https://github.com/aws/aws-encryption-sdk-c/blob/master/source/edk.c#L44
+void aws_cryptosdk_edk_list_clean_up(struct aws_array_list *encrypted_data_keys) {
+    assert(aws_cryptosdk_edk_list_is_valid(encrypted_data_keys));
+    aws_array_list_clean_up(encrypted_data_keys);
+}
+
+/**
+ * Receives encryption request from user and attempts to generate encryption materials,
+ * including an encrypted data key and a list of EDKs for doing encryption.
+ *
+ * On success returns AWS_OP_SUCCESS and allocates encryption materials object at address
+ * pointed to by output.
+ *
+ * On failure returns AWS_OP_ERR, sets address pointed to by output to NULL, and sets
+ * internal AWS error code.
+ */
+int generate_enc_materials(
+    struct aws_cryptosdk_cmm *cmm,
+    struct aws_cryptosdk_enc_materials **output,
+    struct aws_cryptosdk_enc_request *request) {
+    // assert(aws_cryptosdk_cmm_base_is_valid(cmm));
+    // assert(__CPROVER_w_ok(output, sizeof(*output)));
+    // assert(aws_cryptosdk_enc_request_is_valid(request));
```

Comment:
Why these preconditions are commented out? They should be added to the harness.

Reply:
fixed